### PR TITLE
Adds in the E-Tank, a new energy drink!

### PR DESCRIPTION
**Pull request by CRITAWAKETS**
## About The Pull Request

This PR adds in the new hottest energy drink, the E-Tank. It restores 10 stamina damage per sip! It may also make you really drunk in the process... It is made with pina colada and some teslium for some real energy.

img

## Why It's Good For The Game

A new interesting drink, and a good reference while at it.

## Changelog
:cl:
add: A new drink is added to the bartender's arsenal with the E-Tank. Made with 2 units of pina colada and 1 unit of teslium, it sure restores some energy.
/:cl:

Side note, it does nothing on robots. This codebase hates borgs.

Also, the recipe is from a sidequest in Megaman Starforce 3 where you make an alcoholic drink for a sixth grader.

### CODE REVIEW

**Comment by MrDoomBringer** [comment]
image
gang rise up

**Comment by CRITAWAKETS** [comment]
> gang rise up

Dont think i wouldnt make a metroid reference eventually, those games are the ones i grew up with.

**Comment by wesoda25** [comment]
We don’t need anymore combat chems

**Comment by CRITAWAKETS** [comment]
> We don’t need anymore combat chems

someone didnt check just how alcoholic this thing is

it could make a quick escape but you will be screwed afterwards by the alcohol. If its not boozey enough i can make it stronger.

**Comment by wesoda25** [comment]
I didn't see that, so thanks for pointing it out. Still, it won't matter since you made this apply on initial ingestion, not on metabolism. People can simply take like 5 sips to clear all of their stamina damage, and then pop a calomel/antihol pill to dodge the alcohols affects.

**Comment by itseasytosee** [comment]
The tricky thing your making. Either a meme no one will use or something viable. Either way, someone is gonna be mad.

**Comment by MrDoomBringer** [comment]
is that the sound i hear of people picking the heavy drinker quirk to gain combat benefits I hear?

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 1]
```diff
@@ -2268,3 +2268,26 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	description = "A weirdly named energy drink. Tastes tropical and refills your energy instantly!"
+	boozepwr = 75 //robots may not feel it but this is STRONG
+	color = "#5555FF"
+	quality = DRINK_FANTASTIC //atleast its a good drink
+	taste_description = "tropical energy"
+	glass_icon_state = "etank"
+	glass_name = "E Tank"
+	glass_desc = "A weirdly named energy drink can. Tastes tropical and refills your energy instantly!"
+
+/datum/reagent/consumable/ethanol/e_tank/expose_mob(mob/living/M, methods=INGEST, reac_volume, show_message = 1) //thanks synthflesh for being one of the few things in the game with this proc
```
Comment (Jared-Fogle) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:2283]:
```suggestion
/datum/reagent/consumable/ethanol/e_tank/expose_mob(mob/living/M, methods=INGEST, reac_volume, show_message = TRUE) //thanks synthflesh for being one of the few things in the game with this proc
```

**Comment by bluetshirtguy** [comment]
bartender powercreep bad

**Comment by CRITAWAKETS** [comment]
wait i just realized, i think pills count for hte ingestion

dont merge until i make this metabolized because thats just rife with potential for abuse

**Comment by ATH1909** [comment]
vape go brrrrrrr

**Comment by CRITAWAKETS** [comment]
Alright you can merge this now. It uses metabolization so it isnt instant but you also cant cheese it via pentetic or antihol.

**Comment by wesoda25** [comment]
How does this compare to other stamina damage reducing chems?

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 2]
```diff
@@ -2268,3 +2268,20 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	quality = DRINK_FANTASTIC //atleast its a good drink
+	taste_description = "tropical energy"
+	glass_icon_state = "etank"
+	glass_name = "E Tank"
+	glass_desc = "A weirdly named energy drink can. Tastes tropical and refills your energy!"
+	metabolization_rate = 2.5 * REAGENTS_METABOLISM //goes away quickly, 1u for 2 stamina heals.
+
+/datum/reagent/consumable/ethanol/e_tank/on_mob_life(mob/living/carbon/C)
+	C.adjustStaminaLoss(-2*REM, 0)
+	to_chat(C, "<span class='danger'>You feel your energy restored!</span>")
```
Comment (wesoda25) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:2286]: This is going to be spamming chat every cycle, I think. You should probably just remove it.

Comment (CRITAWAKETS) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:2286]: that is... true

Im sorry little one.

Comment (wesoda25) [reply code/modules/reagents/chemistry/reagents/alcohol_reagents.dm]: You could just add a chance for it

**Comment by Krysonism** [comment]
> 
> 
> How does this compare to other stamina damage reducing chems?

It is very weak. The only ones that are worth using are aranesp, eldritch essence / unholy water and maybe turbo.

**Comment by cacogen** [comment]
It looks like a pipe from Super Mario shrunk lengthwise and recoloured blue with a straw from Earthbound

**Comment by MrDoomBringer** [comment]
@cacogen the sprite is lifted almost directly from megaman you cashew

**Review by Twaticus — CHANGES_REQUESTED** [review]
the pitch black outline is too much

i never played megaman so it doesnt get a nostalgia pass sorry!

**Comment by CRITAWAKETS** [comment]
sorry man, but its intentional

i even have a no black outline sprite made, actually looked kinda worse than with black outlines

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 3]
```diff
@@ -2268,3 +2268,19 @@ All effects don't start immediately, but rather get worse over time; the rate is
...
+	boozepwr = 90 //robots may not feel it but this is STRONG
+	color = "#5555FF"
+	quality = DRINK_FANTASTIC //atleast its a good drink
+	taste_description = "tropical energy"
+	glass_icon_state = "etank"
+	glass_name = "E Tank"
+	glass_desc = "A weirdly named energy drink can. Tastes tropical and refills your energy!"
+	metabolization_rate = 2.5 * REAGENTS_METABOLISM //goes away quickly, 1u for 2 stamina heals.
+
+/datum/reagent/consumable/ethanol/e_tank/on_mob_life(mob/living/carbon/C)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:2284]: life needs return value `. = ..()` and life should be living

File: code/modules/reagents/chemistry/reagents/alcohol_reagents.dm [hunk 4]
```diff
@@ -2268,3 +2268,19 @@ All effects don't start immediately, but rather get worse over time; the rate is
 /datum/reagent/consumable/ethanol/pruno/on_mob_life(mob/living/carbon/M)
 	M.adjust_disgust(5)
 	..()
+
+/datum/reagent/consumable/ethanol/e_tank
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/alcohol_reagents.dm:2272]: `/improper` for names?

**Comment by CRITAWAKETS** [comment]
i've gotten way too sick to work on anything, will later update

**Pull request closed without merging**